qcodes/instrument_drivers/stanford_research/SIM928.py

```diff
@@ -0,0 +1,264 @@
+import logging
+import numpy as np
+from functools import partial
+import time
+from qcodes.instrument.visa import VisaInstrument
+from qcodes.utils import validators as vals
+from qcodes.instrument.parameter import ManualParameter
+
+
+class SIM928(VisaInstrument):
+    """
+    A driver for Stanford Research Systems SIM 928 DC source modules installed
+    in a SIM900 mainframe.
+    """
+
+    def __init__(self, name, address, **kw):
+        super().__init__(name, address=address, terminator='\n', **kw)
+
+        self.write('*DCL')  # device clear
+        self.write('FLSH')  # flush port buffers
+        self.write('SRST')  # SIM reset (causes 100 ms delay)
+        time.sleep(0.5)
+
+        self.modules = self.find_modules()
+        for i in self.modules:
+            self.write_module(i, 'TERM LF')
+            self.add_parameter('IDN{}'.format(i),
+                               label="IDN of module {}".format(i),
+                               get_cmd=partial(self.get_idn, i))
+            self.add_parameter('voltage{}'.format(i), unit='V',
```

Comment:
We usually give the modules names, such as `top_gate` etc. Perhaps a way to also give custom names to the SIM modules would be useful? For instance, an optional kwarg that is a dict with items of the form `slot: name`?